### PR TITLE
Feature/clean bump

### PR DESCRIPTION
**Pull request by guillaumevernieres**
have a look at the feature/clean_bump of soca-bundle to build this branch.

From @benjaminmenetrier :

new 4D interfaces with unstructured grid,
better handling of 2D grids for unstructured grids,
BUMP parameters estimation and Dirac tests.

I cleaned up a few issues in the tests, 3dvar tests are ok now, but I still need to fix a few failures elsewhere and speed up some of the tests.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -37,10 +37,14 @@ set( SOCA_LINKER_LANGUAGE CXX )
...
 ################################################################################
+# fckit
+ecbuild_use_package( PROJECT fckit VERSION 0.4.1 REQUIRED )
+include_directories( ${FCKIT_INCLUDE_DIRS} )
+
 # Boost
 set( Boost_MINIMUM_VERSION "1.47" )
-find_package( Boost ${Boost_MINIMUM_VERSION} REQUIRED
-COMPONENTS unit_test_framework )
+#find_package( Boost ${Boost_MINIMUM_VERSION} REQUIRED
```
Comment (aerorahul) [inline CMakeLists.txt]: Please remove these.

**Comment by guillaumevernieres** [comment]
@aerorahul , @travissluka : I'm getting failures in "linearmodel" and "model" test, mpi is finailzed "too early". All the new tests that @benjaminmenetrier added are failing in bump (trying to allocate already allocated array). The rest of the tests are OK.
Lets try to merge that branch without fixing the failures mentioned above, we'll address those in separate pr's. We're quite behind and there's a few more disruptive oops/ufo merges on there way ...

**Comment by guillaumevernieres** [comment]
Tests that should be ignored for this pr:
The following tests FAILED:
	  9 - test_soca_linearmodel (Failed)
	 12 - test_soca_model (Failed)
	 21 - test_soca_parameters_bump_cov_lct (Failed)
	 22 - test_soca_parameters_bump_cov_nicas (Failed)
	 23 - test_soca_dirac_bump_cov (Failed)
	 29 - test_soca_4dhybrid (Failed)

**Comment by travissluka** [comment]
@guillaumevernieres are you testing with the latest `develop` for `fckit`? (it's not set to update in the bundle CMakeLists.txt). I'm getting failures for `test_soca_forecast` and everything after it due to 

`3: /home/tsluka/work/jedi-soca/soca.build.container/soca/test/test_soca_forecast: symbol lookup error: /home/tsluka/work/jedi-soca/soca.build.container/lib/libsoca.so: undefined symbol: __fckit_kdtree_module_MOD_kdtree_create`

**Comment by travissluka** [comment]
... actually, i'm getting that error even in the oops tests with anything that uses bump

**Comment by guillaumevernieres** [comment]
Yes I'm using the latest develop for fckit, I'll add UPDATE to the bundle. You might be using an older version of an already installed fckit instead of the one you build with the bundle? 
I'll give it a go on Theia tomorrow or later tonight, if the motivation is still there ...

**Comment by travissluka** [comment]
... nevermind, I updated the container and it seems to be working. (I can't use it outside the container right now because FMS fails to build due to MPI issues, something wrong with my own config.)

**Comment by guillaumevernieres** [comment]
Just forced push an update to 2 yaml files (don't ask) ... sorry, you need to update, or ensvariance and genpert will fail.

**Comment by travissluka** [comment]
I'm getting the following failures
```
The following tests FAILED:
	  6 - test_soca_enspert (Failed)
	  9 - test_soca_linearmodel (Failed)
	 12 - test_soca_model (Failed)
	 20 - test_soca_dirac_soca_cov (Failed)
	 21 - test_soca_parameters_bump_cov_lct (Failed)
	 22 - test_soca_parameters_bump_cov_nicas (Failed)
	 23 - test_soca_dirac_bump_cov (Failed)
	 29 - test_soca_4dhybrid (Failed)
```
in addition to the ones you said should fail: 

with test 6 The norms are different for me. They are consistent between runs, so it doesn't seem to be a random number issue, perhaps there's a difference with the number of cores used to run?

with test 20, it segfaults and complains `fields:dirac invalid ixdir`

**Comment by travissluka** [comment]
so, yes, with `test_soca_enspert` the answers are consistent between runs, but change with number of cores used

**Comment by guillaumevernieres** [comment]
Ouch, annoying but that makes sense. To make it work we'd need to randomize the global field and broadcast the subdomain to each pe ... or we go back to hard coding the number of pe's for the test.

**Comment by travissluka** [comment]
@guillaumevernieres that's fine for now. We'll need to take a closer look at our tests at some point and get a lower-res model configuration that will run the tests faster.

**Comment by guillaumevernieres** [comment]
@travissluka , Sounds good to me.

**Comment by travissluka** [comment]
The `dirac` yaml parameters are dependent on the number of processors. 
Should we just assume that ctests need to be run with 2 PE for now, and fix these issues in subsequent PRs?

**Comment by guillaumevernieres** [comment]
@travissluka : Yes, we probably should. @benjaminmenetrier was talking about changing the way the dirac tests are done, maybe passing a list of lon/lat and distributing the work properly to the pe ...

**Review by travissluka — APPROVED** [review]

**Pull request merged**